### ISSUE TITLE
Adding #no_nil to an enum segfaults the compiler (instead of reporting an error)

### ISSUE THREAD

**Issue by dogue** [odin-lang/Odin#3355]
## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* Operating System & Odin Version: Linux 6.7.9-arch1-1, Odin version dev-2024-03
* Please paste `odin report` output:

```
        Odin:    dev-2024-03
        OS:      Arch Linux, Linux 6.7.9-arch1-1
        CPU:     AMD Ryzen 5 3600 6-Core Processor
        RAM:     32015 MiB
        Backend: LLVM 14.0.6
```

## Expected Behavior

I would expect the compiler to report an error for an invalid attribute.

## Current Behavior

The compiler experiences a segmentation fault when attempting to compile the example code below.

## Failure Information (for bugs)

Please help provide information about the failure if this is a bug. If it is not a bug, please remove the rest of this template.

### Steps to Reproduce

Here is a complete example to reproduce:

```odin
package no_nil_enum

import "core:fmt"

Enum :: enum #no_nil {A, B, C}

main :: proc() {
	fmt.println("Hellope")
}
```

### Failure Logs

```
✦ ~/Projects/Odin/nonilenum
❯ odin build .
zsh: segmentation fault (core dumped)  odin build .
```

**Comment by dogue** [comment]
Fairly minor issue in the grand scheme of things and entirely my fault. Troubleshooting was as frustrating as you'd expect though.

**Comment by gingerBill** [comment]
This has been fixed in b47d73c65
